adapter/service/outlvlswitch/service.go

```diff
@@ -160,3 +174,37 @@ func (s *service) SetBinaryState(value bool) error {
 
 	return nil
 }
+
+// StartLevelTransition implements starting of the transition with validations and concurrent safety.
+func (s *service) StartLevelTransition(value string, startLvl int, duration time.Duration) error {
+	s.lock.Lock()
+	defer s.lock.Unlock()
+
+	ctr, ok := s.controller.(LevelTransitionController)
+	if !ok {
+		return fmt.Errorf("failed to cast controller into LevelTransitionController when starting level transition")
+	}
+
+	if err := ctr.StartLevelTransition(value, startLvl, duration); err != nil {
```

Comment:
Please add parameter validation before passing them to a method:
* value: up or down
* startLvl: between min_lvl and max_lvl

Please don't forget to reflect that in the tests as well.